storm-core/src/jvm/org/apache/storm/trident/Stream.java

```diff
@@ -124,6 +124,31 @@ public Stream parallelismHint(int hint) {
     }
 
     /**
+     * Sets the CPU Load resource for the current node
+     */
+    public Stream setCPULoad(Number load) {
```

Comment:
> I can look at adding a method for including a component -> resources map when creating a trident topology.
It would be best if we could have these settings as a part of the topology than the stream.

> I may be missing the benefit here, but as far as I can tell, it doesn't read any cleaner for users, The Stream api would have only this one extra non-stream-operation method. E.g. when the user browses the javadocs or when the IDE pops up the methods of the stream.

>and IMO, users shouldn't really need to be aware of Nodes at all. They're supposed to be internal to Trident.
Yes, its ideal if its part of the topology configuration. In my opinion if we are exposing config via the stream api, its better to let it standout than attempting to hide it in with the other streaming operations because the user needs to be aware of what he is doing. `nodeConfig` was only a suggestion, it could be named appropriately.